library/src/test/java/com/novoda/downloadmanager/NetworkRequestFixtures.java

```diff
@@ -0,0 +1,62 @@
+package com.novoda.downloadmanager;
+
+import java.util.Hashtable;
+import java.util.Map;
+
+class NetworkRequestFixtures {
+
+    private Map<String, String> headers = new Hashtable<>();
+    private String url = "http://www.google.com";
+    private NetworkRequest.Method method = NetworkRequest.Method.GET;
+
+    static NetworkRequestFixtures aNetworkRequest() {
+        return new NetworkRequestFixtures();
+    }
+
+    NetworkRequestFixtures withHeader(String key, String value) {
+        headers.put(key, value);
+        return this;
+    }
+
+    NetworkRequestFixtures withUrl(String url) {
+        this.url = url;
+        return this;
+    }
+
+    NetworkRequestFixtures withMethod(NetworkRequest.Method method) {
+        this.method = method;
+        return this;
+    }
+
+    NetworkRequest build() {
+        return new StubNetworkRequest(headers, url, method);
+    }
+
+    private static class StubNetworkRequest implements NetworkRequest {
```

Comment:
You may want to add `toString` and `equals` if you want the `networkRequest` to work properly in tests. You'll find it really hard to debug otherwise when something goes wrong.